securesystemslib/metadata.py

```diff
@@ -120,63 +123,67 @@ def sign(self, signer: Signer) -> Signature:  # pragma: no cover
 
         return signature
 
-    def verify(self, key: Key) -> bool:  # pragma: no cover
-        """verify the payload with provided verifier
-
-        Arguments:
-            key: A "Key" class instance
-
-        Raises:
-            TypeError: If "key" is not a instance of "Key" class.
-
-        Returns:
-            # To be confirmed.
-        """
-
-        if not isinstance(key, Key):
-            raise TypeError(f"expected type 'Key', got {type(key).__name__}")
-
-        signature = None
-        for signature in self.signatures:
-            if signature.keyid == key.keyid:  # what if keyid is none?
-                break
-        else:
-            raise exceptions.NotFoundError(f"No signature found for key '{key.keyid}'")
-
-        valid = key.verify(signature, self.pae)
-        return valid
-
-    def verify_multiple(self, keys: KeyList, c: int) -> bool:  # pragma: no cover
+    def verify(self, keys: KeyList, threshold: int) -> List[str]:  # pragma: no cover
         """verify payload with multiple verifiers
 
         Arguments:
             keys: A list of a "str" and "Key" class instance.
-            c: Condition for cardinality of keys.
+            threshold: required no of signatures have to pass the verification.
 
         Raises:
             TypeError: If key in "keys" is not a instance of "Key" class.
+            ValueError: If "threshold" is not valid.
+            SignatureVerificationError: If "recognized_signers" are less than provided
+                threshold.
 
         Returns:
-            # To be confirmed.
+            recognized_signers: set of "name" of "keys" for which verification succeeds.
```

Comment:
```suggestion
            recognized_signers: list of key names for which verification succeeds.
```
I think you meant `list` here as a `set` has a different meaning - a data structure guaranteeing the uniqueness of the elements inside it.